query-engine/query-engine-c-abi/src/engine.rs

```diff
@@ -0,0 +1,667 @@
+                // This version of the query engine supports connecting via Rust bindings directly
+                // support for JS drivers can be added, but I commented it out for now
+                let connector_kind = ConnectorKind::Rust {
+                    url,
+                    datasource: data_source,
+                };
+
+                let executor = load_executor(connector_kind, preview_features).await?;
+                let connector = executor.primary_connector();
+
+                let conn_span = tracing::info_span!(
+                    "prisma:engine:connection",
+                    user_facing = true,
+                    "db.type" = connector.name(),
+                );
+
+                connector.get_connection().instrument(conn_span).await?;
+
+                Result::<_>::Ok(executor)
+            };
+
+            let query_schema_span = tracing::info_span!("prisma:engine:schema");
+            let query_schema_fut = tokio::runtime::Handle::current()
+                .spawn_blocking(move || {
+                    let enable_raw_queries = true;
+                    schema::build(arced_schema_2, enable_raw_queries)
+                })
+                .instrument(query_schema_span);
+
+            let (query_schema, executor) = tokio::join!(query_schema_fut, executor_fut);
+
+            Ok(ConnectedEngine {
+                schema: builder.schema.clone(),
+                query_schema: Arc::new(query_schema.unwrap()),
+                executor: executor?,
+                engine_protocol: builder.engine_protocol,
+                native: ConnectedEngineNative {
+                    config_dir: builder.native.config_dir.clone(),
+                    env: builder.native.env.clone(),
+                    #[cfg(all(not(target_os = "ios"), not(target_os = "android")))]
+                    metrics: None,
+                },
+            }) as Result<ConnectedEngine>
+        }
+        .instrument(span)
+        .await?;
+
+        *inner = Inner::Connected(engine);
+        Ok(())
+    }
+
+    pub async fn query(
+        &self,
+        body_str: *const c_char,
+        trace_str: *const c_char,
+        tx_id_str: *const c_char,
+    ) -> Result<String> {
+        let dispatcher = self.logger.dispatcher();
+
+        async move {
+            let inner = self.inner.read().await;
+            let engine = inner.as_engine()?;
+
+            let body = get_cstr_safe(body_str).expect("Prisma engine execute body is missing");
+            let tx_id = get_cstr_safe(tx_id_str);
+            let trace = get_cstr_safe(trace_str).expect("Trace is needed");
+
+            let query = RequestBody::try_from_str(&body, engine.engine_protocol())?;
+
+            let span = if tx_id.is_none() {
+                tracing::info_span!("prisma:engine", user_facing = true)
+            } else {
+                Span::none()
+            };
+
```

Comment:
```suggestion
            let span = tracing::info_span!("prisma:engine", user_facing = true);
```

See https://github.com/prisma/prisma-engines/pull/4336